### PR TITLE
Use ProseMirror Sub-views to implement editability of Vue-component slot content

### PR DESCRIPTION
**Pull request by enjibby**
Please understand the the code in its current strate is very much a work in progress, but I will be attempting to improve this to a point where things make a little more sense.

This includes the pre-cursor work to implement the inner-view code from [this example](https://prosemirror.net/examples/footnote/).

### CODE REVIEW

**Comment by enjibby** [comment]
I will be working on this to implement what I can of #868

**Comment by enjibby** [comment]
These changes are the first attempt at implementing the example code outlined in #868 and the beginnings of updating the ComponentView to support it. I expect that I will need to implement a ComponentSlotView for each of the editable sub-areas, but I haven't fully mapped this out yet. Fingers crossed!

File: examples/Components/Routes/VueSlots/StaffList/DefaultBlockSlotExtension.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      },
+      selectable: false,
+      draggable: false,
+      parseDOM: [{
+        tag: 'template[v-slot\\:default]',
+        getAttrs: dom => ({
+          'v-slot:default': dom.getAttribute('v-slot:default'),
+        }),
+      }],
+      toDOM: node => ['template', node.attrs, 0],
```
Comment (kepta) [inline examples/Components/Routes/VueSlots/StaffList/DefaultBlockSlotExtension.js:23]: I am not familiar with Vue, but wouldn't this clash with the DOM [template](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template).

**Comment by hanspagel** [comment]
Thanks for working on this! It’s still on our list to look at subviews and at your implementation. I suppose your work will help us to wrap our heads around it.

That said, don’t expect us to merge that into tiptap 1. We try put all our time into the new version and won’t add new features to tiptap 1.

Thanks for your understanding!

**Comment by enjibby** [comment]
I'd already expected that tiptap V2 was the focus here and I imagine that there would differences, so no issue there. I'm still to prove this would even work given possible incompatibilities in individual browsers.

I will continue to press on when I am able 🙂

File: examples/Components/Routes/VueSlots/StaffList/DefaultBlockSlotExtension.js (hunk 1 above)
Comment (enjibby) [reply examples/Components/Routes/VueSlots/StaffList/DefaultBlockSlotExtension.js:23]: I'm not familiar enough with the inner workings of vue to give you an answer on that myself, but Vue does make use of `<template>` elements within its document structure.

I have used this element type successfully in Vue apps using tiptap by injecting the resulting output of `getHTML()` (that includes a `<template>` element) into a Vue component as so:

```
<component :is="injectedDocString"></component>
```

**Pull request closed without merging**